segmentation/utils/synthetic_spatial_datagen.py

```diff
@@ -222,10 +221,302 @@ def generate_test_label_map(size_img=(1024, 1024), num_A=100, num_B=100, num_C=1
 
     # now generate the sample xarray
     sample_img = np.zeros((1, size_img[0], size_img[1], 1)).astype(np.int16)
-    sample_img[0, :, :, 0] = copy.deepcopy(label_mat)
+    sample_img[0, :, :, 0] = deepcopy(label_mat)
     sample_img_xr = xr.DataArray(sample_img,
                                  coords=[[1], range(size_img[0]), range(size_img[1]), ['segmentation_label']],
                                  dims=['fovs', 'rows', 'cols', 'channels'])
 
     # and return the xarray to pass into calc_dist_matrix, plus the centroid_indices to readjust it
     return sample_img_xr, centroid_indices
+
+
+def generate_two_cell_ring_coords(size_img, center_1, center_2, outer_radius, inner_radius):
+    """
+    This function generates the coordinates for a ring by generating circle perimeters in the desired range of radii
+
+    Arguments:
+        size_img (tuple): the dimensions of the image we wish to generate
+        center_1 (tuple): the center of the first cell
+        center_2 (tuple): the center of the second cell
+        outer_radius (int): the radius of the outer disk of the desired ring
+        inner_radius (int): the radius of the inner disk of the desired ring
+
+    Returns:
+        ring_region_1 (tuple): a tuple indicating the coordinates for the ring we wish to generate for cell 1
+        ring_region_2 (tuple): similar to ring_region_1 for cell 2
+    """
+
+    # create numpy arrays to hold the resulting coordinates
+    ring_region_1_x = np.array([]).astype(np.int16)
+    ring_region_1_y = np.array([]).astype(np.int16)
+    ring_region_2_x = np.array([]).astype(np.int16)
+    ring_region_2_y = np.array([]).astype(np.int16)
+
+    # for each radius in the range of the inner and outer radii
+    for rad in range(inner_radius, outer_radius + 1):
+        # generate circle perimeters for region 1 and 2
+        region_1_perim_x, region_1_perim_y = circle_perimeter(center_1[0], center_1[1], rad, shape=size_img)
+        region_2_perim_x, region_2_perim_y = circle_perimeter(center_2[0], center_2[1], rad, shape=size_img)
+
+        # now add the coordinates to the respective numpy array
+        ring_region_1_x = np.concatenate((ring_region_1_x, region_1_perim_x))
+        ring_region_1_y = np.concatenate((ring_region_1_y, region_1_perim_y))
+        ring_region_2_x = np.concatenate((ring_region_2_x, region_2_perim_x))
+        ring_region_2_y = np.concatenate((ring_region_2_y, region_2_perim_y))
+
+    # create a tuple for the x and y coordinates for each cell ring
+    ring_region_1 = (ring_region_1_x, ring_region_1_y)
+    ring_region_2 = (ring_region_2_x, ring_region_2_y)
+
+    return ring_region_1, ring_region_2
```

Comment:
I feel like we can just return the image; I don't see a real need to return tuples of coordinates